packages/snap/src/helpers/credential-helpers.ts

```diff
@@ -245,6 +248,57 @@ export class VCManager {
     );
   }
 
+  public async clearDuplicates(credential: W3CCredential) {
+    const client = this.ceramicProvider.client();
+
+    const ownerDid = client.did?.id;
+
+    if (!ownerDid) {
+      throw new TypeError('Client not authenticated');
+    }
+
+    const queryHash = hashVC(
+      String(credential.credentialSubject.type),
+      credential.issuer,
+      ownerDid,
+    );
+
+    const hashedOwnerDid = this.personalHashStr(client.did.id);
+
+    const hashedQueryHash = this.personalHashStr(queryHash);
+
+    const data = await loadAllCredentialsListPages<
+      GetVerifiableCredentialsByQueryHashQueryVariables,
+      GetVerifiableCredentialsByQueryHashQuery
+    >(
+      GetVerifiableCredentialsByQueryHash,
+      {
+        first: 1000,
+        queryHash: hashedQueryHash,
+        ownerDid: hashedOwnerDid,
+      },
+      this.ceramicProvider,
+    );
+
+    const ids = data
+      .map((el) => el.verifiableCredentialIndex?.edges)
+      .flat()
+      .map((el) => el?.node?.id ?? '')
+      .filter((el) => Boolean(el));
+
+    await Promise.all(
```

Comment:
Do you just delete all credentials matching the query? I'd rename the method if that's true